### ISSUE TITLE
FineGrainedAuthorizationHandler called twice and fails on the second call

### ISSUE THREAD

**Issue by MattRuwe** [Hawxy/Fga.Net#13]
First of all, thank you so much for creating this library.  It is awesome!

When I initially installed Fga.Net I had the authorization misconfigured.  Not only was I adding authorization via the `services.AddAuthorization` call, but I was redundantly adding it via the `AddControllers` method like this:

```csharp
builder.Services.AddControllers((mvcOptions =>
{
    mvcOptions.Filters.Add(new AuthorizeFilter());
}));
```
Doing this has the unintended and difficult to debug effect of causing the `FineGrainedAuthorizationHandler` Authorization Handler to be called twice.  The first call is successful, but the second calls fails because the `Context.Resource` value that is passed in is not a `HttpContext`.  Instead the object type is a `AuthorizationFilterContextSealed` which I assume is an internal MSFT class.

While I definitely had my API misconfigured, I wonder if it would be better to throw an error in this situation.  At the very least, it would've been helpful to have some kind of warning log to indicate the problem.  As it was, I had to download the code, compile locally and debug into it from my project to figure out the problem.

**Comment by Hawxy** [comment]
Hiya, I'm working on a new release at the moment so I'll update the handler to throw an exception if the incorrect `Resource` type gets passed in. 

Out of interest, was this a project previously updated from .NET Core 2.x? `AuthorizeFilter` is relic of the pre-endpoint routing era and I'm surprised it hasn't been marked as obsolete.

**Comment by MattRuwe** [comment]
No, it was a new project, but I copied code from an old one which is how I got into trouble.  Thanks for looking at this!

**Comment by Hawxy** [comment]
Fixed in v1.1

**Comment by MattRuwe** [comment]
Awesome!  Thank you!
